pkg/descheduler/core/helper.go

```diff
@@ -62,26 +63,52 @@ func NewSchedulingResultHelper(binding *workv1alpha2.ResourceBinding) *Schedulin
 func (h *SchedulingResultHelper) FillUnschedulableReplicas(unschedulableThreshold time.Duration) {
 	reference := &h.Spec.Resource
 	undesiredClusters, undesiredClusterNames := h.GetUndesiredClusters()
+	// clusterIndexToEstimatorPriority key refers to index of cluster slice, value refers to the EstimatorPriority of who gave its estimated result
+	clusterIndexToEstimatorPriority := make(map[int]estimatorclient.EstimatorPriority)
+
 	// Set the boundary.
 	for i := range undesiredClusters {
 		undesiredClusters[i].Unschedulable = math.MaxInt32
 	}
-	// Get the minimum value of MaxAvailableReplicas in terms of all estimators.
+
+	// get estimators and sort it by priority.
 	estimators := estimatorclient.GetUnschedulableReplicaEstimators()
+	sort.Slice(estimators, func(i, j int) bool {
+		return estimators[i].GetPriority() > estimators[j].GetPriority()
+	})
+
+	priorityOfLastAvailableEstimator := estimatorclient.EstimatorPriority(math.MinInt32)
 	ctx := context.WithValue(context.TODO(), util.ContextKeyObject,
 		fmt.Sprintf("kind=%s, name=%s/%s", reference.Kind, reference.Namespace, reference.Name))
+
 	for _, estimator := range estimators {
+		// if higher-priority estimators have formed a full result of member clusters, no longer to call lower-priority estimator.
+		if estimator.GetPriority() < priorityOfLastAvailableEstimator && len(clusterIndexToEstimatorPriority) == len(undesiredClusterNames) {
+			break
+		}
 		res, err := estimator.GetUnschedulableReplicas(ctx, undesiredClusterNames, reference, unschedulableThreshold)
 		if err != nil {
 			klog.Errorf("Max cluster unschedulable replicas error: %v", err)
 			continue
 		}
+		priorityOfLastAvailableEstimator = estimator.GetPriority()
 		for i := range res {
+			// the result of this cluster estimated failed, ignore the corresponding result
 			if res[i].Replicas == estimatorclient.UnauthenticReplica {
 				continue
 			}
-			if undesiredClusters[i].ClusterName == res[i].Name && undesiredClusters[i].Unschedulable > res[i].Replicas {
+			// the cluster name not match, ignore, which hardly ever happens
+			if res[i].Name != undesiredClusters[i].ClusterName {
+				continue
```

Comment:
If not matched, we need to add logs to locate the problem because something goes wrong in the estimator.

Reply:
got it